src/snmalloc/ds/pagemap.h

```diff
@@ -59,6 +59,14 @@ namespace snmalloc
     {
       SNMALLOC_ASSERT(is_initialised());
 
+      if constexpr (has_bounds)
+      {
+        if ((p - base >= size) || (p + length - base >= size))
+        {
+          p = p - base;
+        }
+      }
+
```

Comment:
I think it would be good to have an error in the case where we try to register a range that is not part of the fixed range.

Reply:
In the case of a fixed range, it is currently guaranteed to be out of range.
If that is not supposed to happen, then yes, I would agree.
But then there's an upstream fix to be made somewhere :)

Reply:
Alright, you were absolutely correct :)

I didn't test properly, and the check was a classic off-by-one.

The end-address should be checked with > and not with >=, as it's of course perfectly legal to end on the last byte of the arena.

Reply:
One thing that is still weird is that although it now works, when using fixed range our application uses a lot more memory.

I suspect though that's unrelated to this specific thing, so I'll dig a bit and open a new PR when I have something

Reply:
Also, the crashes on some platforms are... odd
I don't have access to those platforms though, so they're a bit hard to reproduce :(

Reply:
> Also, the crashes on some platforms are... odd I don't have access to those platforms though, so they're a bit hard to reproduce :(

Which platforms are crashing?  I have got a clean CI run, so don't know what needs investigating. Could you let me know what they are?

Reply:
Ah, it was in the previous runs it seems. They were crashing on Mac and some older Linux, but not the newer.
I have no idea why :)

But yes, they're all green now. Great!